docs(tutorials): sentry (#1334)
scaleway/docs-content: tutorials/sentry-error-tracking/index.mdx

---
meta:
  title: Configuring Sentry error tracking
  description: Discover how to configure a Sentry monitoring & error tracking server on a Scaleway Instance.
content:
  h1: Configuring Sentry error tracking
  paragraph: Discover how to configure a Sentry monitoring & error tracking server on a Scaleway Instance.
tags: compute messaging apache Sentry error error-tracking Instance
categories:
  - compute
  - instances
hero: assets/scaleway_sentry.webp
dates:
  validation: 2023-02-21
  posted: 2018-06-07
---

## Sentry - Overview

Today's world is running on applications and everyone needs visibility into application health - no matter if you are an engineer, support or product manager. Sentry is an open source solution for monitoring & error tracking in applications. It is written in Python and can provide real-time reporting of errors in applications based on a client/server architecture. You can easily integrate it into your applications via its SDK and let them automatically report the errors via common messaging tools like Slack, IRC and more. This helps reduce the time for development of your applications and increases the user experience.

> 

Author: ```suggestion
<Message type="tip">
  We recommend you follow this tutorial using a [Cost-Optimized Instance](/compute/instances/reference-content/choosing-instance-type/).
</Message>
```

---
meta:
  title: Configuring Sentry error tracking
  description: Discover how to configure a Sentry monitoring & error tracking server on a Scaleway Instance.
content:
  h1: Configuring Sentry error tracking
  paragraph: Discover how to configure a Sentry monitoring & error tracking server on a Scaleway Instance.
tags: compute messaging apache Sentry error error-tracking Instance
categories:
  - compute
  - instances
hero: assets/scaleway_sentry.webp
dates:
  validation: 2023-02-21
  posted: 2018-06-07
---

## Sentry - Overview

> Today's world is running on applications and everyone needs visibility into application health - no matter if you are an engineer, support or product manager. Sentry is an open source solution for monitoring & error tracking in applications. It is written in Python and can provide real-time reporting of errors in applications based on a client/server architecture. You can easily integrate it into your applications via its SDK and let them automatically report the errors via common messaging tools like Slack, IRC and more. This helps reduce the time for development of your applications and increases the user experience.

Reviewer: ```suggestion
Today's world is running on applications and whether you are an engineer, support technician or product manager, everyone needs visibility into application health. Sentry is an open source solution for monitoring & error tracking in applications. It is written in Python and can provide real-time reporting of errors in applications based on a client/server architecture. You can easily integrate it into your applications via its SDK and let them automatically report the errors via common messaging tools like Slack, IRC and more. This helps reduce the time for development of your applications and improves the user experience.
```

<Message type="tip">
  We recommend you follow this tutorial using a [Cost-Optimized Instance](/compute/instances/reference-content/choosing-instance-type/).

> </Message>

Author: ```suggestion
</Message>

```

You can find all reports on a dashboard which makes it easy to triage the problem - how often it occurs, which impact it has on the user experience of your application, which part of your code causes the problem and so on.

> This tutorial details how to setup a [Sentry](https://sentry.io/welcome/) Self-Hosted server on a Scaleway compute Instance to start error tracking in your applications.

Reviewer: ```suggestion
This tutorial details how to set up a [Sentry](https://sentry.io/welcome/) Self-Hosted server on a Scaleway Instance to start error tracking in your applications.
```

<Macro id="iam-requirements" />
 
<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [configured your SSH Key](/console/my-project/how-to/create-ssh-key/)

>   - You have a [Instance](https://www.scaleway.com/en/virtual-instances/) running on Ubuntu Jammy Jellyfish (22.04 LTS)

Reviewer: ```suggestion
  - You have an [Instance](https://www.scaleway.com/en/virtual-instances/) running on Ubuntu Jammy Jellyfish (22.04 LTS)
```

</Message>

<Message type="note">
  Sentry requires at least 8 GB of RAM to run. We recommend, therefore, that you use a Scaleway Instance with at least 4 CPU cores, 8 GB of RAM, and 20 GB of available disk space to follow this tutorial.
</Message>

## Installing Sentry

In this tutorial, we will install Sentry on an [Ubuntu Jammy Jellyfish (22.04 LTS)](https://www.ubuntu.com) based Instance.

> 1. Update the system and install the required prerequisites.

Reviewer: ```suggestion
1. Update the system and install the necessary prerequisites.
```

    ```
    apt update && apt upgrade -y
    apt install apt-transport-https ca-certificates certbot curl git gnupg-agent nginx python3-certbot-nginx sendmail software-properties-common ufw -y
    ```

> 2.  Sentry runs in a Docker container using `docker-compose`. Install the GPG Key of Docker.

Reviewer: ```suggestion
2.  Install Docker's GPG key, necessary as Sentry runs in a Docker container using `docker-compose`.
```

---
meta:
  title: Configuring Sentry error tracking
  description: Discover how to configure a Sentry monitoring & error tracking server on a Scaleway Instance.
content:
  h1: Configuring Sentry error tracking
  paragraph: Discover how to configure a Sentry monitoring & error tracking server on a Scaleway Instance.
tags: compute messaging apache Sentry error error-tracking Instance
categories:
  - compute
  - instances
hero: assets/scaleway_sentry.webp
dates:
  validation: 2023-02-21
  posted: 2018-06-07
---

## Sentry - Overview

> Today's world is running on applications and whether you are an engineer, support technician or product manager, everyone needs visibility into application health. Sentry is an open source solution for monitoring & error tracking in applications. It is written in Python and can provide real-time reporting of errors in applications based on a client/server architecture. You can easily integrate it into your applications via its SDK and let them automatically report the errors via common messaging tools like Slack, IRC and more. This helps reduce the time for development of your applications and improves the user experience.

Author: ```suggestion
In today's fast-paced world, applications are the lifeblood of many industries. From engineers to support technicians to product managers, everyone needs to stay on top of their application's health to ensure smooth operation. That's where Sentry comes in. This powerful open-source solution is designed to monitor and track errors in applications, helping you identify issues before they become major problems. Built in Python, Sentry boasts a client/server architecture that provides real-time error reporting. And with its SDK, integrating Sentry into your applications is a breeze. You can even set it up to report errors automatically via popular messaging platforms like Slack and IRC. With Sentry by your side, you can streamline your development process, improve your applications, and give your users the seamless experience they crave.
```

    ```
2.  Install Docker's GPG key, necessary as Sentry runs in a Docker container using `docker-compose`.
    ```
    sudo mkdir -m 0755 -p /etc/apt/keyrings
    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
    ```
3.  Configure Docker's software repositories to be able to install the required software on your Instance.
    ```
    echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
    $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
    ```
4. Update the apt package manager and install Docker Engine and the required plugins using `apt`.
    ```
    apt update && apt install docker-ce docker-ce-cli docker-compose containerd.io docker-buildx-plugin docker-compose-plugin
    ```
5. Enter the `/opt` directory on your Instance and download the [latest release](https://github.com/getsentry/self-hosted/releases/latest) of Self-Hosted Sentry.
    ```
    cd /opt
    wget https://github.com/getsentry/self-hosted/archive/refs/tags/23.2.0.tar.gz
    ```
    <Message type="note">

>       At the time of writing of this tutorial, the latest released version was `23.2.0`.

Reviewer 2: ```suggestion
      At the time this tutorial was written, the latest released version was `23.2.0`.
```
